langlib/lang.string/src/main/java/org/ballerinalang/langlib/string/PadStart.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.langlib.string;
+
+import io.ballerina.runtime.api.creators.ErrorCreator;
+import io.ballerina.runtime.api.utils.StringUtils;
+import io.ballerina.runtime.api.values.BString;
+
+import static io.ballerina.runtime.api.constants.RuntimeConstants.STRING_LANG_LIB;
+import static io.ballerina.runtime.internal.util.exceptions.BallerinaErrorReasons.LENGTH_GREATER_THAT_2147483647_NOT_YET_SUPPORTED;
+import static io.ballerina.runtime.internal.util.exceptions.BallerinaErrorReasons.getModulePrefixedReason;
+
+/**
+ * Extern function ballerina.model.strings:padStart.
+ *
+ * @since 2201.1.0
+ */
+
+public class PadStart {
+
+    public static BString padStart(BString str, long len, BString padChar) {
+        int strLength = str.length();
```

Comment:
Should we not consider returnning immediately if the padChar is an empty string? we don't need to go through all the logic below right? please check for padEnd also

Reply:
in fact, if the padChar is empty and the len is larger than the str length, then we can never add padding to satisfy the length. How does python handle this?

Reply:
Empty string is not belonging to `string:Char`. Therefore, it should be a compile time error.
```ballerina
string:Char x = ""; // incompatible types: expected 'string:Char', found 'string'
```
I have added few tests for such scenarios
L34, L53, L72 in https://github.com/ballerina-platform/ballerina-lang/pull/35598/files#diff-52b72539d21e3d57324fd2d52853587aeb1e2ad70fa93efec05ad83fa177bf54